crates/driver/src/infra/notify/notification.rs

```diff
@@ -0,0 +1,46 @@
+use crate::domain::competition;
+
+/// A notification is sent to the solvers in case a solution failed validation.
+#[derive(Debug)]
+pub struct Notification {
+    pub auction_id: Option<competition::auction::Id>,
+    pub kind: Kind,
+}
+
+#[derive(Debug)]
+pub enum Kind {
+    /// The solution doesn't contain any user orders.
+    EmptySolution, // NoUserOrders,
+    /// The solution violated a price constraint (ie. max deviation to external
+    /// price vector)
+    PriceViolation,
+    /// No valid score could be computed for the solution.
+    ScoringFailed,
+    /// Solution aimed to internalize untrusted token/s.
+    UntrustedInternalization, /* NonBufferableTokensUsed */
+    /// Solver don't have enough balance to submit the solution.
+    InsufficientBalance, /* /// The solution didn't pass simulation. Includes all data
```

Comment:
Wouldn't this be caught in the simulation already?

Reply:
Added todo to https://github.com/cowprotocol/services/issues/1891. I hope you agree removing this is optimization and not blocking colocation prod.